android/app/src/main/kotlin/com/jabook/app/jabook/audio/processors/SpeechCompressorAudioProcessor.kt

```
@@ -0,0 +1,357 @@
+            output.put(input)
+            return
+        }
+        val channels = format.channelCount
+        val samples = input.remaining() / (2 * channels)
+        applySpeechCompression(input, output, samples, channels)
+    }
+
+    /**
+     * 3-band speech compression: crossover filters → per-band compression → sum → output.
+     */
+    private fun applySpeechCompression(
```

detekt (1.23.8)

[warning] The function applySpeechCompression appears to be too complex based on Cognitive Complexity (complexity: 25). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)